Microsoft.Toolkit.HighPerformance/Helpers/NullableHelper.cs

```diff
@@ -0,0 +1,35 @@
+using System;
+using System.Runtime.CompilerServices;
+
+
+namespace Microsoft.Toolkit.HighPerformance.Helpers
+{
+    /// <summary>
+    /// Helpers to perform operations on nullable value types
+    /// </summary>
+    public static class NullableHelper
+    {
+        /// <summary>
+        /// We use this type to publically expose the fields of a Nullable{T} type
+        /// </summary>
+        private struct NullableHelperMap<T>
```

Comment:
This mapping is only guaranteed to be correct on .NET Core runtimes.
The entire file should be inside an `#if NETCORE_RUNTIME` compiler directive. Look at the `NullableRef<T>` file to see an example. This could end badly on eg. Mono or Unity, or webassembly, for instance. Remember that this package also targets .NET Standard 1.4 and up 😄

Also the name should follow the pattern of other mapping types in the package:
```suggestion
        private struct RawNullableData<T>
```